lib/app_localizations.dart

```diff
@@ -0,0 +1,58 @@
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+class AppLocalizations {
+
+  AppLocalizations(this.locale);
+  final Locale locale;
+  static AppLocalizations of(BuildContext context) {
+    return Localizations.of<AppLocalizations>(context, AppLocalizations);
+  }
+
+  static const LocalizationsDelegate<AppLocalizations> delegate =
+      _AppLocalizationsDelegate();
+
+  Map<String, String> _localizedStrings;
+
+  Future<bool> load() async {
+    final String jsonString =
+        await rootBundle.loadString('assets/locale/${locale.languageCode}.json');
+    final Map<String, dynamic> jsonMap = json.decode(jsonString);
+
+    _localizedStrings = jsonMap.map((key, value) {
+      return MapEntry(key, value.toString());
+    });
+
+    return true;
+  }
+
+  String translate(String key) {
+    if(_localizedStrings.containsKey(key)){
+      return _localizedStrings[key];
+    }
+    return '';
```

Comment:
Do you think it makes sense to throw an exception if we are not able to find a localized string in the bundle when it should actually be present since we are specifically looking for it.

Reply:
This change has been done